File: examples/Simple.cpp
#include "sprng.h"
#include "sprng_cpp.h"

void printRandInts(Sprng *rng, int num)
{
   int i;
   rng->print_sprng();
   for (i = 0; i < num; i++)
   {
      printf ("%d\n", rng->isprng());
   }
}

void printRandDoubles(Sprng *rng, int num)
{
   int i;
   rng->print_sprng();
   for (i = 0; i < num; i++)
   {
      printf ("%.16lf\n", rng->sprng());
   }
}

int main (void)
{
   int seed = 42;
   Sprng * gen1, **gens;
   gen1 = SelectType(SPRNG_LFG);
   gen1->init_sprng(0, 1, seed, 0);
   gen1->spawn_sprng(3, &gens);
   printRandInts(gen1, 10);
   printRandDoubles(gen1, 10);
   printRandInts(gens[0], 10);
   printRandDoubles(gens[0], 10);
   printRandInts(gens[1], 10);
   printRandDoubles(gens[1], 10);
   printRandInts(gens[2], 10);
   printRandDoubles(gens[2], 10);
   return 0;
}

File: examples/Simple.hs
module Main where

import System.Random.SPRNG.LFG
import Control.Monad (replicateM_)

main = do
   let seed = 42
   gen1 <- create seed
   [gen2, gen3, gen4] <- spawn gen1 3
   printRandInts gen1 10
   printRandDoubles gen1 10
   printRandInts gen2 10
   printRandDoubles gen2 10
   printRandInts gen3 10
   printRandDoubles gen3 10
   printRandInts gen4 10
   printRandDoubles gen4 10

printRandInts :: Gen -> Int -> IO ()
printRandInts rng num =
   replicateM_ num (print =<< (uniform rng :: IO Int))

printRandDoubles :: Gen -> Int -> IO ()
printRandDoubles rng num =
   replicateM_ num (print =<< (uniform rng :: IO Double))
